rust/tool/src/generation.rs

```diff
@@ -96,7 +95,7 @@ impl fmt::Display for Generation {
 }
 
 fn read_build_time(path: &Path) -> Result<Date> {
-    let build_time = time::OffsetDateTime::from_unix_timestamp(fs::metadata(path)?.mtime())?.date();
+    let build_time = time::OffsetDateTime::from(fs::metadata(path)?.created()?).date();
```

Comment:
I think you don't need all this, try:
```suggestion
    let build_time: time::OffsetDateTime = fs::metadata(path)?.created()?.into();
```
I would assume that a proper `Into<T>` impl is in the context.

Reply:
Probably: `Ok(fs::metadata(path)?.created()?.into())` would work too.

Reply:
Thanks for the feedback.

Sure, I'll try this.
I was trying to get this to work with minimal changes, so I didn't want to modify anything outside getting the date to work.